File: selectionsort.c
/*This is an implimentation of selectionsort algorithm*/
/*kksante*/
/*21JAN15*/

#include<stdio.h>


void selectionsort(int A[], int n) {
        int i, j;
                for(i = 0; i < n-1; i++) { // we need to do n-2 passes
                        int iMin = i; /*ith position: elements from 
                                        i till n-1 are candidates*/
                        for(j = i+1; j < n; j++) {
                                if (A[j] < A[iMin])
                                        iMin = j; /*update the index
                                                    of min element*/
                        }
                int temp = A[i];
                A[i] = A[iMin];
                A[iMin] = temp;
        }
}

int main() 
{
        int A[11] = {23, 494, 1, 3, 586, 234234, 2, 9, 6, 8, 6};
        selectionsort(A, 11);
        int i;
        for(i = 0; i<11; i++) {
                printf ("%d ", A[i]);
        }
        return 0;
}

File: selectionsort.py
#!/usr/bin/env python

# Implimentation of selection sort algorithm
# kksante
# 21JAN15
# The Selection Sort
import random

# Create a list of random numbers
list = []
for i in range(11):
    list.append(random.randrange(100))


def selection_sort(list):
    # Loop through the entire array
    for curPos in range(len(list)):
        # Find the position that has the smallest element
        # Start with the current position
        minPos = curPos

        # Scan left to right (end of the list)
        for scanPos in range(curPos + 1, len(list)):

            # Is this position smallest?
            if list[scanPos] < list[minPos]:

                # It is, mark this position as the smallest
                minPos = scanPos

        # Swap the two values
        temp = list[curPos]
        list[curPos] = list[minPos]
        list[minPos] = temp

# Code to print list


def print_list(list):
    for item in list:
        print "{:3}".format(item),
    print()

# Try out the sort
print_list(list)
selection_sort(list)
print_list(list)
